Add `Beatmap submission (lazer)` (#14900)
ppy/osu-wiki: wiki/Beatmapping/Beatmap_submission_(lazer)/en.md

# Beatmap submission (lazer)

*For the osu!(stable) version of this article, see: [Beatmap submission](/wiki/Beatmapping/Beatmap_submission)*

> [Beatmaps](/wiki/Beatmap) can be submitted to the osu! website via the [in-game editor](/wiki/Client/Beatmap_editor). Submitting a beatmap allows it to receive attention from other users and potentially enter the [Ranked](/wiki/Beatmap/Category#ranked) or [Loved](/wiki/Beatmap/Category#loved) categories. This is done through the a process commonly called the **Beatmap Submission System** (***BSS***).

Reviewer: At some point, the `/wiki/Client/Beatmap_editor` should be linking to lazer beatmap editor instead.

Author: added todo comment for this

![The beatmap submission screen](img/bss.png "Beatmap submission")

> Selecting `Submit beatmap` from the `File` dropdown in the editor (shortcut: `Ctrl` + `Shift` + `U`) will open the *beatmap submission* screen. This screen provides options for which [beatmap category](/wiki/Beatmap/Category) the map should end up in, whether to enable discussion notifications and whether open the uploaded map in the browser. If you encounter trouble when using BSS, please ask in the [Help](https://osu.ppy.sh/community/forums/5) subforum.

Reviewer 2: ```suggestion
Selecting `Submit beatmap` from the `File` dropdown in the editor (shortcut: `Ctrl` + `Shift` + `U`) will open the *beatmap submission* screen. This screen provides options for which [beatmap category](/wiki/Beatmap/Category) the map should end up in, whether to enable discussion notifications and whether to open the uploaded map in the browser. If you encounter trouble when using BSS, please ask in the [Help](https://osu.ppy.sh/community/forums/5) subforum.
```

# Beatmap submission (lazer)

*For the osu!(stable) version of this article, see: [Beatmap submission](/wiki/Beatmapping/Beatmap_submission)*

> [Beatmaps](/wiki/Beatmap) can be submitted to the osu! website via the [in-game editor](/wiki/Client/Beatmap_editor)<!-- TODO: link lazer editor article when available -->. Submitting a beatmap allows it to receive attention from other users and potentially enter the [Ranked](/wiki/Beatmap/Category#ranked) or [Loved](/wiki/Beatmap/Category#loved) categories. This is done through the a process commonly called the **Beatmap Submission System** (***BSS***).

Reviewer 3: a the an
I also have a longstanding impression that the BSS is, well, a system, and not the process itself, but don't know how to phrase it better — "using the component ..."?
```suggestion
[Beatmaps](/wiki/Beatmap) can be submitted to the osu! website via the [in-game editor](/wiki/Client/Beatmap_editor)<!-- TODO: link lazer editor article when available -->. Submitting a beatmap allows it to receive attention from other users and potentially enter the [Ranked](/wiki/Beatmap/Category#ranked) or [Loved](/wiki/Beatmap/Category#loved) categories. This is done through the process commonly called the **Beatmap Submission System** (***BSS***).
```

Author: do you mean something like this? i would indeed call bss a system too but i was just trying to refer to it without writing "system" twice

```suggestion
[Beatmaps](/wiki/Beatmap) can be submitted to the osu! website via the [in-game editor](/wiki/Client/Beatmap_editor)<!-- TODO: link lazer editor article when available -->. Submitting a beatmap allows it to receive attention from other users and potentially enter the [Ranked](/wiki/Beatmap/Category#ranked) or [Loved](/wiki/Beatmap/Category#loved) categories. The infrastructure facilitating this is commonly called the **Beatmap Submission System** (***BSS***).
```

Reviewer 3: I thiiink so

![The beatmap submission screen](img/bss.png "Beatmap submission")

> Selecting `Submit beatmap` from the `File` dropdown in the editor (shortcut: `Ctrl` + `Shift` + `U`) will open the *beatmap submission* screen. This screen provides options for which [beatmap category](/wiki/Beatmap/Category) the map should end up in, whether to enable discussion notifications and whether to open the uploaded map in the browser. If you encounter trouble when using BSS, please ask in the [Help](https://osu.ppy.sh/community/forums/5) subforum.

Reviewer 3: don't think the italic is needed here
```suggestion
Selecting `Submit beatmap` from the `File` dropdown in the editor (shortcut: `Ctrl` + `Shift` + `U`) will open the beatmap submission screen. This screen provides options for which [beatmap category](/wiki/Beatmap/Category) the map should end up in, whether to enable discussion notifications and whether to open the uploaded map in the browser. If you encounter trouble when using BSS, please ask in the [Help](https://osu.ppy.sh/community/forums/5) subforum.
```

## Limitations

<!-- reference: https://github.com/ppy/osu-server-beatmap-submission/blob/b52dc670d8361b0f25ec2a2edf016398142cfb21/osu.Server.BeatmapSubmission/BeatmapSubmissionController.cs -->

> Beatmaps will fail to submit if they exceed the online file size or difficulty limit. The file size limit is 5 MiB plus an additional 10 MiB for every minute of beatmap length, and it caps at 200 MB<!-- reference: https://github.com/ppy/osu-server-beatmap-submission/blob/b52dc670d8361b0f25ec2a2edf016398142cfb21/osu.Server.BeatmapSubmission/Program.cs#L21 -->. The difficulty limit is currently 128.

Reviewer 3: gotta choose one, and while MiB may be proper, MB is more widespread and easily recognisable on the internets, especially in a not-so-technical conversation

```suggestion
Beatmaps will fail to submit if they exceed the online file size or difficulty limit. The file size limit is 5 MB plus an additional 10 MB for every minute of beatmap length, and it caps at 200 MB<!-- reference: https://github.com/ppy/osu-server-beatmap-submission/blob/b52dc670d8361b0f25ec2a2edf016398142cfb21/osu.Server.BeatmapSubmission/Program.cs#L21 -->. The difficulty limit is currently 128.
```

Author: conflating MB with MiB is a common misnomer that would just be misleading here. we really do mean 5 * 1024^2 bytes, 10 * 1024^2 bytes and 200 * 1000^2 bytes, and i dunno about converting to decimal MBs

Reviewer 3: what I'm saying is that most of the time whenever someone brings up `MB` they really do mean `MiB` (especially in case of storage units, unless they are a damn storage manufacturer) and that's what ~everyone goes by and recognizes anyway

if you want to be really precise, then sure, let's set a trend, but I think this will need to be addressed everywhere then

(including the article on BSS in stable)

Author: i hope i'm not the only one who would get confused by this... (have applied)
